src/harness/testcases/WINNF_FT_S_GRA_testcase.py

```diff
@@ -997,3 +999,217 @@ def test_WINNF_FT_S_GRA_16(self):
+    # into the PPA zone
+    for device_index in ppa_cluster_list:
+      devices[device_index]['installationParam']['latitude'], devices[
+          device_index]['installationParam'][
+              'longitude'] = getRandomLatLongInPolygon(ppa_record)
+    for device_index in devices_in_ppa_zone:
+      devices[device_index]['installationParam']['latitude'], devices[
+          device_index]['installationParam'][
+              'longitude'] = getRandomLatLongInPolygon(ppa_record)
+
+    config = {
+        'registrationRequest': devices,
+        'conditionalRegistrationData': conditionals,
+        'ppa': {
+            'ppaRecord': ppa_record,
+            'palRecords': pal_record,
+            # Indexing the same way as CBSD IDs.
+            'ppaClusterList': ppa_cluster_list
+        },
+        'grantRequest': grant_requests,
+        'expectedResponseCodes': [
+            (0,),  # all valid params => SUCCESS
+            (102,),  # missing CBSD ID => MISSING_PARAM
+            (103,),  # LowFrequency > HighFrequency => INVALID_VALUE
+            (300,),  # partially overlapping CBRS band => UNSUPPORTED_SPECTRUM
+            (400,),  # CBSD inside (but not part of) claimed PPA => INTERFERENCE
+        ]
+    }
+    writeConfig(filename, config)
+
+  @configurable_testcase(generate_GRA_17_default_config)
+  def test_WINNF_FT_S_GRA_17(self, config_filename):
+    """[Configurable] Array grant request."""
+
+    config = loadConfig(config_filename)
+    # Very light checking of the config file.
+    self.assertEqual(
+        len(config['registrationRequest']),
+        len(config['grantRequest']))
+    self.assertEqual(
+        len(config['grantRequest']),
+        len(config['expectedResponseCodes']))
+
+    # Whitelist FCC IDs.
+    for device in config['registrationRequest']:
+      self._sas_admin.InjectFccId({
+          'fccId': device['fccId'],
+          'fccMaxEirp': 47
+      })
+
+    # Whitelist user IDs.
+    for device in config['registrationRequest']:
+      self._sas_admin.InjectUserId({'userId': device['userId']})
+
+    # Pre-load conditional registration data for N3 CBSDs.
+    if ('conditionalRegistrationData' in config) and (
+        config['conditionalRegistrationData']):
+      self._sas_admin.PreloadRegistrationData(
+          config['conditionalRegistrationData'])
+
+    # Inject PAL database record
+    for pal_record in config['ppa']['palRecords']:
+      self._sas_admin.InjectPalDatabaseRecord(pal_record)
+
+    # Register devices
+    cbsd_ids = self.assertRegistered(config['registrationRequest'])
+
+    # Update PPA record with device's CBSD ID and Inject zone data
+    for device_index in config['ppa']['ppaClusterList']:
+      config['ppa']['ppaRecord']['ppaInfo']['cbsdReferenceId'] = [
+          cbsd_ids[device_index]
+      ]
+    zone_id = self._sas_admin.InjectZoneData({
+        'record': config['ppa']['ppaRecord']
+    })
+    self.assertTrue(zone_id)
+
```

Comment:
Missing trigger CPAS if N2 PPAs >= 0.
  if ('ppa' in config) and (config['ppa']): # and possibly (len(config['ppa']] > 0)
    # Trigger daily activities
    self.TriggerDailyActivitiesImmediatelyAndWaitUntilComplete()

Reply:
Fixed.